### PR TITLE
Move and update v1beta1 to v1 migration doc from Indexed

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes

This commit follows up on #5415 and resolves the respective examples and docs on the migration.

Fixes: #5330 
/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [n/a] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
The document has been added that describes the changes needed when migrating from v1beta1 to v1.
```

### CODE REVIEW

File: docs/migrating-v1beta1-to-v1.md [hunk 1]
```diff
@@ -74,6 +74,15 @@ to `Pipeline`s and `Task`s defined in other namespaces in the cluster.
...
 apiVersion: tekton.dev/v1beta1
 kind: TaskRun
+metadata:
+  name: cluster-task-reference
+spec:
+  taskRef:
+    name: example-task
+    kind: ClusterTask
+---
+apiVersion: tekton.dev/v1
```
Comment (lbernick) [inline docs/migrating-v1beta1-to-v1.md:84]: I'd suggest:

before:
```yaml
apiVersion: tekton.dev/v1beta1
kind: TaskRun
metadata:
  name: cluster-task-reference
spec:
  taskRef:
    name: example-task
    kind: ClusterTask
```
After:
...

This example makes it look like you need to use both at the same time

Comment (lbernick) [reply docs/migrating-v1beta1-to-v1.md:84]: same for the bundle one above

File: docs/migrating-v1beta1-to-v1.md [hunk 2]
```diff
@@ -49,7 +49,7 @@ spec:
     name: example-task
     bundle: python:3-alpine
 ---
-apiVersion: tekton.dev/v1beta1
+apiVersion: tekton.dev/v1
 kind: Taskrun
```
Comment (lbernick) [inline docs/migrating-v1beta1-to-v1.md:53]:
```suggestion
kind: TaskRun
```

Comment (lbernick) [reply docs/migrating-v1beta1-to-v1.md:53]: same above as well

**Review by vdemeester — COMMENTED** [review]
Can we also move this document elsewhere for the time being ? 
If we were to release tomorrow, a new page would appear ([Migrating from v1beta1](https://tekton.dev/vault/pipelines-main/migrating-v1beta1-to-v1/)) which would be a bit confusing as it would make it look like `v1` is already available, when it is not yet the case.

**Comment by JeromeJu** [comment]
> Can we also move this document elsewhere for the time being ? If we were to release tomorrow, a new page would appear ([Migrating from v1beta1](https://tekton.dev/vault/pipelines-main/migrating-v1beta1-to-v1/)) which would be a bit confusing as it would make it look like `v1` is already available, when it is not yet the case.

@vdemeester May I ask where would be a good place to put it or shall I just remove it at the moment ?

**Comment by vdemeester** [comment]
> > Can we also move this document elsewhere for the time being ? If we were to release tomorrow, a new page would appear ([Migrating from v1beta1](https://tekton.dev/vault/pipelines-main/migrating-v1beta1-to-v1/)) which would be a bit confusing as it would make it look like `v1` is already available, when it is not yet the case.
> 
> @vdemeester Thanks for the comment. May I ask where would be a good place to put it or shall I just remove it at the moment ?

Hum we definitely shouldn't remove it. Maybe in a `dev/docs` or anywhere that is not "indexed" by our website.

**Comment by lbernick** [comment]
Created https://github.com/tektoncd/pipeline/issues/5478 for the test flakes on this PR

**Review by vdemeester — APPROVED** [review]
/lgtm

**Pull request merged**